apps/aecore/src/aec_pow_cuckoo.erl

```diff
@@ -181,6 +187,43 @@ verify(Hash, Nonce, Soln) ->
             {error, {unknown, {C, E}}}
     end.
 
+%%------------------------------------------------------------------------------
+%% @doc
+%%   Creates the Cuckoo buffer (hex encoded) from a base64-encoded hash and a
+%%   uint64 nonce.
+%% @end
+%%------------------------------------------------------------------------------
+-spec pack_header_and_nonce(binary(), aec_pow:nonce()) -> string().
+pack_header_and_nonce(Hash, Nonce) ->
+    %% Cuckoo originally uses 32-bit nonces inserted at the end of its 80-byte buffer.
+    %% This buffer is hashed into the keys used by the main algorithm.
+    %%
+    %% We insert our 64-bit Nonce right after the hash of the block header We
+    %% base64-encode both the hash of the block header and the nonce and pass
+    %% the resulting command-line friendly string with the -h option to Cuckoo.
+    %%
+    %% The SHA256 hash is 32 bytes (44 chars base64-encoded), the nonce is 8 bytes
+    %% (12 chars base64-encoded). That leaves plenty of room (80 - 56 = 24
+    %% bytes) for cuckoo to put its nonce (which will be 0 in our case) in.
+    %%
+    %% (Base64 encoding: see RFC 3548, Section 3:
+    %% https://tools.ietf.org/html/rfc3548#page-4
+    %% converts every triplet of bytes to 4 characters: from N bytes to 4*ceil(N/3)
+    %% bytes.)
+    %%
+    %% Like Cuckoo, we use little-endian for the nonce here.
+    NonceStr = base64:encode_to_string(<<Nonce:64/little-unsigned-integer>>),
+    HashStr = base64:encode_to_string(Hash),
```

Comment:
May I suggest hardcoding expectations on lengths of base64-encoded NonceStr and HashStr here please? It would make the code more readable.

Reply:
Guard on pack_header_and_nonce clause head would help too.